### PR TITLE
Refactor pulp::child and use puppetlabs-apache vhost options

### PR DESCRIPTION
**Pull request by ekohl**
This needs testing. I'm mostly wondering what you think of the pattern.

I also noticed that pulp::child has unused parameters which makes me think this is not usuable as a stand alone class while the rest implies it does.

### CODE REVIEW

File: manifests/child/apache.pp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    directories            => $directories,
+    ssl                    => true,
+    ssl_cert               => $ssl_cert,
+    ssl_key                => $ssl_key,
+    ssl_ca                 => $ssl_ca,
+    ssl_verify_client      => 'optional',
+    ssl_options            => '+StdEnvVars',
+    ssl_verify_depth       => '3',
+    # allow older yum clients to connect, see bz 647828
+    custom_fragment        => 'SSLInsecureRenegotiation On',
```
Comment (stbenjam) [inline manifests/child/apache.pp]: I would guess EL5.  There's still a lot of CentOS 5 boxes out there - how about making it parameterized and default off?

Comment (ekohl) [inline manifests/child/apache.pp]: That sounds fine to me.

**Comment by ekohl** [comment]
I'll revisit this when https://github.com/Katello/puppet-katello/pull/189 has been merged.

File: manifests/child.pp [hunk 2]
```diff
@@ -4,22 +4,18 @@
...
-  $parent_qpid_scheme    = 'ssl',
-  $parent_qpid_port      = '5671',
-  $oauth_effective_user  = 'admin',
-  $oauth_key             = 'key',
-  $oauth_secret          = 'secret',
-  $ssl_cert              = '/etc/pki/pulp/ssl_apache.crt',
-  $ssl_key               = '/etc/pki/pulp/ssl_apache.key',
-  $server_ca_cert        = '/etc/pki/pulp/ca.crt'
+  String[1] $parent_fqdn,
+  String $oauth_effective_user = 'admin',
```
Comment (ekohl) [inline manifests/child.pp]: Note I left these in, but they are *not* used. ```pulp::child::config``` uses values from the main pulp class. I'm wondering if that's a bug

**Comment by ekohl** [comment]
I'm wondering where this code is even used. I can't really find where this is included in the installer.

**Comment by stbenjam** [comment]
We don't use `pulp::child` in Katello, all the pulps are basically masters since 3.1 I think.  It's still a feature of pulp that people may want to use though.

File: manifests/child.pp (hunk 2 above)
Comment (stbenjam) [inline manifests/child.pp]: It may be, I think pulp::child is intended to be used standalone so it should use it's own values.

**Comment by ekohl** [comment]
I'm leaning to merging this and changing things in incompatible ways in another PR.

**Comment by ekohl** [comment]
Updated: added a trailing comma in child.pp.

**Pull request merged**